packages/foam-core/src/model/tags.ts

```diff
@@ -0,0 +1,159 @@
+import { FoamWorkspace } from './workspace';
+import { URI } from './uri';
+import { IDisposable } from '../index';
+import { Resource } from './note';
+
+export type TagMetadata = { uri: URI };
+
+export class FoamTags implements IDisposable {
+  public readonly tags: { [key: string]: TagMetadata[] } = {};
```

Comment:
a very good point that comes from https://github.com/foambubble/foam/pull/729:
we should use `Map` instead of plain objects.

Otherwise a `#constructor` tag would break the code - can you check that is the case?

Reply:
I've changed it to a `Map`.